### ISSUE TITLE
OOM no where explained

### ISSUE THREAD

**Issue by santosh0705** [docker/docker#14399]
I couldn't find documentation on OOM. What is this all about?, how and when the OOM event is generated? etc etc...

**Comment by vielmetti** [comment]
Look here:

http://docs.docker.com/reference/run/

e.g.

> By default, Docker kills processes in a container if an out-of-memory (OOM) error occurs. To change this behaviour, use the --oom-kill-disable option. Only disable the OOM killer on containers where you have also set the -m/--memory option. If the -m flag is not set, this can result in the host running out of memory and require killing the host’s system processes to free memory.

**Comment by santosh0705** [comment]
I noticed the oom event is occurring when I'm stopping the container even I started only bash shell and there is no other processes are running. A bash shell doesn't consume much memory than why this oom?

**Comment by phemmer** [comment]
To be clear on the provided documentation, docker doesn't kill anything in an out-of-memory event, the kernel does.
It also may not be your container which is using memory, it could be anything on the host. Things inside, or outside the container. All the `--oom-kill-disable` option does is prevent the kernel from picking things inside your container when it needs to kill something to get more memory.

For more details, these 2 articles seem to do a good job from a high(ish) level:
https://www.kernel.org/doc/gorman/html/understand/understand016.html
https://plumbr.eu/blog/memory-leaks/out-of-memory-kill-process-or-sacrifice-child

Basically something on your box is sucking up memory (but not necessarily your container), you need to find out what.

**Comment by santosh0705** [comment]
thanks @phemmer 
Got a clear view of OOM.
Im testing docker in a VM on my Mac and may be OOM activated due to less memory. But still I cannot figure it out why it is happening only when I'm stopping the container, stopping the container is actually cleaning up the occupied memory right. My container are just a bash shell and not doing anything productive or running memory intensive application and I'm running only one container at a time.

Its not happening always, only sometimes its occurs. I'll try to increase the VM memory and test it again.

**Comment by jfrazelle** [comment]
I am going to close this one and defer to @phemmer's in https://github.com/docker/docker/issues/14440
